overlord/configstate/configcore/vitality.go

```diff
@@ -34,13 +39,79 @@ func init() {
 	supportedConfigurations["core."+vitalityOpt] = true
 }
 
-func handleVitalityConfiguration(tr config.ConfGetter, opts *fsOnlyContext) error {
-	// XXX: rewrite/restart changed configs
+func isNotInstalled(err error) bool {
+	_, ok := err.(*snap.NotInstalledError)
+	return ok
+}
+
+func handleVitalityConfiguration(tr config.Conf, opts *fsOnlyContext) error {
+	var pristineVitalityStr, newVitalityStr string
+
+	if err := tr.GetPristine("core", vitalityOpt, &pristineVitalityStr); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if err := tr.Get("core", vitalityOpt, &newVitalityStr); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if pristineVitalityStr == newVitalityStr {
+		return nil
+	}
+
+	st := tr.State()
+	st.Lock()
+	defer st.Unlock()
+
+	oldVitalityMap := map[string]int{}
+	newVitalityMap := map[string]int{}
+	// assign "0" (delete) rank to all old entries
+	for i, instanceName := range strings.Split(pristineVitalityStr, ",") {
+		oldVitalityMap[instanceName] = i + 1
+		newVitalityMap[instanceName] = 0
+	}
+	// build rank of the new entries
+	for i, instanceName := range strings.Split(newVitalityStr, ",") {
+		newVitalityMap[instanceName] = i + 1
+	}
+
+	for instanceName, rank := range newVitalityMap {
+		info, err := snapstate.CurrentInfo(st, instanceName)
+		if isNotInstalled(err) {
+			continue
+		}
+		if err != nil {
+			return err
+		}
+		// nothing rank to do if rank is unchanged
+		if oldVitalityMap[instanceName] == newVitalityMap[instanceName] {
+			continue
+		}
+		// rank changed, rewrite/restart services
+		for _, app := range info.Apps {
+			if !app.IsService() {
+				continue
+			}
+			opts := &wrappers.AddSnapServicesOptions{VitalityRank: rank}
+			if err := wrappers.AddSnapServices(info, nil, opts, progress.Null); err != nil {
+				return err
+			}
+		}
+		// XXX: copied from handlers.go:startSnapServices()
+		svcs := info.Services()
+		startupOrdered, err := snap.SortServices(svcs)
+		if err != nil {
+			return err
+		}
+		tm := timings.New(nil)
+		pb := progress.Null
+		if err = wrappers.StartServices(startupOrdered, pb, tm); err != nil {
```

Comment:
```suggestion
		if err = wrappers.StartServices(startupOrdered,  progress.Null, tm); err != nil {
```

Nitpick, more compact and consistent with similar code above.